src/Tickets/Commerce/Cart/Abstract_Cart.php

```diff
@@ -70,23 +71,74 @@ static function ( $item ) {
 	}
 
 	/**
-	 * Get the total of the cart.
+	 * Get the total value of the cart, including additional values such as fees or discounts.
 	 *
+	 * This method calculates the total by first computing the subtotal from all items in the cart,
+	 * and then applying any additional values (e.g., fees or discounts) provided via the `tec_tickets_commerce_get_cart_total_value` filter.
+	 *
+	 * @since TBD Refactored logic, to include a new filter.
 	 * @since 5.10.0
 	 *
-	 * @return null|float
+	 * @return float The total value of the cart, or null if there are no items.
 	 */
-	public function get_cart_total() {
-		if ( null !== $this->cart_total ) {
-			return $this->cart_total;
+	public function get_cart_total(): float {
+		$subtotal = $this->get_cart_subtotal();
+		if ( ! $subtotal ) {
+			return 0.0;
 		}
 
 		$items = $this->get_items_in_cart( true );
 
+		// Extract subtotals from the cart items.
+		$sub_totals = array_filter( wp_list_pluck( $items, 'sub_total' ) );
+
+		/**
+		 * Filters the total value in the cart to add additional fees or discounts.
+		 *
+		 * Additional values must be instances of the `Value` class to ensure consistent behavior.
+		 *
+		 * @since TBD
+		 *
+		 * @param Value[] $values     An array of `Value` instances representing additional fees or discounts.
+		 * @param array   $items      The items currently in the cart.
+		 * @param Value   $sub_totals The total of the subtotals from the items.
+		 */
+		$additional_values = apply_filters(
+			'tec_tickets_commerce_get_cart_total_value',
+			[],
+			$items,
+			Value::create()->total( $sub_totals )
+		);
+
+		// Combine the subtotals and additional values.
+		$total_value = Value::create()->total( array_merge( $sub_totals, $additional_values ) );
+
+		$this->cart_total = $total_value->get_decimal();
+
+		return $total_value->get_decimal();
+	}
+
+	/**
+	 * Get the subtotal of the cart items.
+	 *
+	 * The subtotal is the sum of all item subtotals without additional values like fees or discounts.
+	 *
+	 * @since TBD Refactored to avoid cumulative calculations.
+	 *
+	 * @return float The subtotal of the cart.
+	 */
+	public function get_cart_subtotal(): float {
+		// Reset cart_total to ensure it's not cumulative across calls.
+		$this->cart_total = 0.0;
+
+		$items = $this->get_items_in_cart( true );
+
+		// If no items in the cart, return null.
 		if ( empty( $items ) ) {
-			return null;
+			return 0.0;
 		}
 
+		// Calculate the total from the subtotals of each item.
 		foreach ( $items as $item ) {
 			$this->cart_total += $item['sub_total']->get_decimal();
```

Comment:
I would assume no since thats how its been working for a while now. If we dont have a sub_total better we get a warning (if not something obvious like an uncaught exception) since we need that to calculate totals